layouts/partials/footer.html

```diff
@@ -1,14 +1,55 @@
 
-      {{ if .Params.chapter }}
+      {{ $isChapter := or (eq .Kind "section") (.Params.chapter)}}
+      {{ if $isChapter }}
         </div> <!-- end chapter-->
       {{ end }}
       </div>
     </div>
 
+    {{ if $.Site.Params.autoNav }}
+    {{ $.Scratch.Set "_sectionpages" (slice) }}
+    {{ if $.Site.Params.menu }}
+      {{ range $sname := $.Site.Params.menu }}
+        {{ $.Scratch.Add "_sectionpages" ($.Site.GetPage "section" $sname) }}
+      {{ end }}
+    {{ else }}
+      {{ range $sname, $spages := .Site.Sections }}
+        {{ $.Scratch.Add "_sectionpages" ($.Site.GetPage "section" $sname) }}
+      {{ end }}
+      {{ $.Scratch.Set "_sectionpages" (sort ($.Scratch.Get "_sectionpages") "Weight") }}
+    {{ end }}
+    {{ $.Scratch.Set "_allpages" (slice) }}
+    {{ range $.Scratch.Get "_sectionpages" }}
+      {{ if .Content }}
+        {{ $.Scratch.Add "_allpages" . }}
+      {{ end }}
+      {{ range index $.Site.Sections .Section }}
+        {{ $.Scratch.Add "_allpages" .Page }}
+      {{ end }}
+    {{ end }}
+    {{ end }}
+
     <div id="navigation">
-        {{ $Site := .Site }}
-        {{ with .Params.prev }}<a class="nav nav-prev" href="{{ $Site.BaseURL }}{{ . }}"> <i class="fa fa-chevron-left"></i></a>{{ end }}
-        {{ with .Params.next }}<a class="nav nav-next" href="{{ $Site.BaseURL }}{{ . }}" style="margin-right: 0px;"><i class="fa fa-chevron-right"></i></a>{{ end }}
+        {{ if $.Site.Params.autoNav }}
+        {{ $allpages := $.Scratch.Get "_allpages" }}
+        {{ range $index, $element := $allpages }}
+          {{ $uniqueid := ($.Scratch.Get "uniqueId" | default $.UniqueID ) }}
+          {{if eq $element.UniqueID $uniqueid }}
+            {{ if not (isset $.Params "prev") }}
+              {{ with index $allpages (sub $index 1) }}
+                <a class="nav nav-prev" href="{{ $.Site.BaseURL }}{{ .URL }}"> <i class="fa fa-chevron-left"></i></a>
```

Comment:
I tested it again, and I have problems when using `baseurl` for arrow links. For example, I use "https://matcornic.github.io/hugo-learn-doc" as `baseurl` in my `config.toml`. So when I `hugo serve`, the server is starting at http://localhost:1313/hugo-learn-doc/

But, when I set `autoNav=true`, `href` of arrows links are not the right ones. For example:

- Current page: http://localhost:1313/hugo-learn-doc/basics/
- Arrow link expected: http://localhost:1313/hugo-learn-doc/basics/what-is-this-hugo-theme/
- Actual arrow link: http://localhost:1313/hugo-learn-doc//hugo-learn-doc/basics/what-is-this-hugo-theme/

It seems we have already the `$.Site.BaseURL` in `.URL`. I removed it locally and it seems to work (when I set `baseurl` or not)

`prev`/`next` parameters overriding have to keep `$.Site.BaseURL` though.